### PR TITLE
Chapter 6

### PR DESCRIPTION
**Pull request by balqis277**
ready

### CODE REVIEW

**Comment by balqis277** [comment]
ready

File: Gemfile [hunk 1]
```diff
@@ -21,7 +21,6 @@ gem "jbuilder", "~> 2.7"
 # Use Redis adapter to run Action Cable in production
 # gem 'redis', '~> 4.0'
 # Use Active Model has_secure_password
-# gem 'bcrypt', '~> 3.1.7'
```
Comment (ngocvt-0484) [inline Gemfile:24]: why you comment it ?

File: test/controllers/users_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+require "test_helper"
+
+class UsersControllerTest < ActionDispatch::IntegrationTest
+  test "should get new" do
```
Comment (ngocvt-0484) [inline test/controllers/users_controller_test.rb:4]: dont need do test, so please delete this file

File: Gemfile (hunk 1 above)
Comment (balqis277) [reply Gemfile:24]: I follow the video tutorial. Should I remove it?

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

File: Gemfile (hunk 1 above)
Comment (huongnt-2545) [reply Gemfile:24]: nope, you need to un comment this line
```
gem 'bcrypt', '~> 3.1.7'
```

File: test/fixtures/users.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (huongnt-2545) [inline test/fixtures/users.yml:1]: delete this file, it's no need

File: test/models/user_test.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/models/user_test.rb:1]: delete this file

**Review by huongnt-2545 — COMMENTED** [review]
This pull request has 2 commits, need to combine in 1
btw, commented

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

File: Gemfile (hunk 1 above)
Comment (anlvq-1058) [reply Gemfile:24]: please  uncomment this line
`gem 'bcrypt', '~> 3.1.7'`

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+    before_save {self.email = email.downcase}
```
Comment (huongnt-2545) [inline app/models/user.rb:2]: indentation = 2 spaces
```suggestion
  before_save {self.email = email.downcase}
```

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+  
+    validates :name, presence: true
+    VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+    validates :email, presence: true, length: {maximum: 255}, format: {with: VALID_EMAIL_REGEX}
+    validates :email, uniqueness: true
+  
+    has_secure_password
+    validates :password, presence: true, length: { minimum: 6 }
+  end
```
Comment (huongnt-2545) [inline app/models/user.rb:11]: add new empty line at the end of file

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

**Review by anlvq-1058 — COMMENTED** [review]
merged

**Pull request merged**